packages/x-date-pickers/src/internals/demo/DemoContainer.tsx

```diff
@@ -152,9 +152,9 @@ export function DemoContainer(props: DemoGridProps) {
       };
     }
   } else if (childrenSupportedSections.has('date-time')) {
-    sx = { ...sx, [`& > .${textFieldClasses.root}`]: { minWidth: 270 } };
+    sx = { ...sx, [`& > .${textFieldClasses.root}`]: { minWidth: 260 } };
```

Comment:
Please correct me if I'm wrong, but it seems that these changes are not necessary and have 0 effect because all of the demos with the `clearable` field have an explicit width set... And after all, having `minWidth` wouldn't solve the issue with `clearable`, because it causes a layout shift anyways. 🙈 

```suggestion
    sx = { ...sx, [`& > .${textFieldClasses.root}`]: { minWidth: 270 } };
```